libs/react-hooks/src/useCurrentResourceId.ts

```diff
@@ -0,0 +1,29 @@
+import { useWebContext } from '@webb-tools/api-provider-environment';
+import { ResourceId } from '@webb-tools/sdk-core';
+import { useEffect, useState } from 'react';
+
+/**
+ * Hook to get the current resource id
+ * @returns The current resource id
+ */
+export const useCurrentResourceId = () => {
+  const [resourceId, setResourceId] = useState<ResourceId | null>(null);
+
+  const { activeApi } = useWebContext();
+
+  // Fetch the current resource id
+  useEffect(() => {
+    if (!activeApi) {
+      return;
+    }
+
+    const fetchResourceId = async () => {
+      const resourceId = await activeApi.getResourceId();
+      setResourceId(resourceId);
+    };
+
+    fetchResourceId();
+  }, [activeApi]);
+
+  return resourceId;
+};
```

Comment:
Consider using the `useCallback` hook to memoize the `fetchResourceId` function to avoid creating a new function reference on every render.

Add error handling to catch and log any errors that occur during the `getResourceId` call.

Reply:
Added the error handling in the `fetchResourceId` function, we cannot use `useCallback` here because the function is inside the `useEffect` hook, and it only runs once the `activeApi` changes.